libs/backend/src/list_directory.ts

```diff
@@ -110,6 +110,35 @@ export async function listDirectory(
   }
 }
 
+/**
+ * Get entries for a directory recursively, includes stat information for each entry.
+ * Requires that the path be absolute. Includes directories in result.
+ */
+export async function listDirectoryRecursive(
```

Comment:
This could end up loading a lot of data in memory at once. Maybe consider something like this, and for now you can just `Array.from(listDirectoryRecursive(…))` and bail if you find an `Err`?

```ts
/**
 * Get entries for a directory recursively, includes stat information for each entry.
 * Requires that the path be absolute. Includes directories in result.
 */
export async function* listDirectoryRecursive(
  path: string
): AsyncGenerator<Result<FileSystemEntry[], ListDirectoryError>> {
  const listRootResult = await listDirectory(path);
  if (listRootResult.isErr()) {
    yield listRootResult;
  } else {
    for (const fileEntry of listRootResult.ok()) {
      if (fileEntry.type === FileSystemEntryType.Directory) {
        yield* listDirectoryRecursive(fileEntry.path);
      }
    }
  }
}
```

Reply:
In the current use case I'm not concerned about memory over-usage but this utility could be used in that way. Just counted the total number of files on my dev machine and it was over a million :). I'll switch to the generator version.

Reply:
OK, I just pushed a change to use the generator version. It seems clunky but perhaps that is the cost of memory efficiency. Take a look?

Reply:
I intend to use the the utils from #3148 once that is merged.